### ISSUE TITLE
Saving some YouTube links leads to 'Cannot read properties of undefined' error in console when valid API key set in settings

### ISSUE THREAD

**Issue by darcydriscoll** [DominikPieper/obsidian-ReadItLater#92]
### Please check that this issue hasn't been reported before.

- [X] I searched previous [Bug Reports](https://github.com/DominikPieper/obsidian-ReadItLater/labels/type%3A%20bug) didn't find any similar reports.

### Expected Behavior

When I run a YouTube link through the plugin, with a valid API key set in settings, I expect it to save into Obsidian. e.g. this link saves successfully: https://www.youtube.com/watch?v=fAq6vFKOcXc

### Current behaviour

When saving some YouTube links, with a valid API key set in settings, the plugin throws an error and does not save the link. For example, this link throws an error: https://www.youtube.com/watch?v=-L32tZMbSZI

This is the error thrown:

```error
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'map')
    at Jt.parseApiResponse (plugin:obsidian-read-it-later:53:15620)
    at async Jt.prepareNote (plugin:obsidian-read-it-later:53:13522)
    at async Et.processClipboard (plugin:obsidian-read-it-later:140:1164)
    at async Object.callback (plugin:obsidian-read-it-later:140:836)
```

This occurs even when the template does not contain any strings that require access to the YouTube API.

### Steps to reproduce

Have the following YouTube note template title in plugin settings:

```
Youtube - %title%
```

Also have the following YouTube note template in plugin settings:

```
# [%videoTitle%](%videoURL%)

## Video

(%videoURL%)

%videoPlayer%

## Description

%videoDescription%

## Notes

- 

#readitlater/unread
```

The %videoDescription% string is optional -- the bug occurs without it.

1. Generate a YouTube API v3 key on Google Cloud
2. Plug the key into the ReadItLater plugin settings
3. Copy this link: https://www.youtube.com/watch?v=-L32tZMbSZI
4. Open Obsidian
5. Open the command palette
6. Execute the command: 'ReadItLater - Save Clipboard'
7. Note the error in the console

### Which Operating Systems are you using?

- [ ] Android
- [ ] iPhone/iPad
- [ ] Linux
- [ ] macOS
- [X] Windows

### Obsidian Version

1.1.16

### Plugin Version

0.2.0

### Checks

- [X] I have tried it with all other plugins disabled and the error still occurs

**Comment by lukacovicadam** [comment]
Thanks for the report. Fix is in #93 and will be included in next release.

**Comment by darcydriscoll** [comment]
Awesome, thanks Adam :)

**Comment by lukacovicadam** [comment]
Released in [0.3.0](https://github.com/DominikPieper/obsidian-ReadItLater/releases/tag/0.3.0)
